File: Servidor_Horacerta/src/main/java/org/example/protocol/Comunicado.java
package org.example.protocol;

import java.io.Serializable;

public abstract class Comunicado implements Serializable {
    private static final long serialVersionUID = 1L;

    @Override
    public String toString() {
        return "Comunicado [Classe: " + this.getClass().getSimpleName() + "]";
    }

    @Override
    public int hashCode() {

        return super.hashCode();
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) return true;

        if (obj == null || getClass() != obj.getClass()) return false;

        return true;
    }

}

File: Aplicativo_HoraCerta/app/src/main/java/com/example/aplicativo_horacerta/network/Comunicado.kt
package com.example.aplicativo_horacerta.network

import java.io.Serializable

// Classe mãe, envia os objetos para o server
open class Comunicado : Serializable {
    companion object {
        // Valor igual ao Comunicado.java
        private const val serialVersionUID = 1L
    }
}